examples/mixed-spmv/mixed-spmv.cpp

```diff
@@ -0,0 +1,271 @@
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+// @sect3{Include files}
+
+// This is the main ginkgo header file.
+#include <ginkgo/ginkgo.hpp>
+
+// Add the fstream header to read from data from files.
+#include <fstream>
+// Add the C++ iostream header to output information to the console.
+#include <iostream>
+// Add the STL map header for the executor selection.
+#include <map>
+// Add the string manipulation header to handle strings.
+#include <string>
+// Add the timing header for timing.
+#include <chrono>
+// Add the random header to generate random vectors.
+#include <random>
+
+namespace {
+
+
+/**
+ * Generate a random value.
+ *
+ * @tparam ValueType  valuetype of the value
+ * @tparam ValueDistribution  type of value distribution
+ * @tparam Engine  type of random engine
+ *
+ * @param value_dist  distribution of array values
+ * @param engine  a random engine
+ *
+ * @return ValueType
+ */
+template <typename ValueType, typename ValueDistribution, typename Engine>
+typename std::enable_if<!gko::is_complex_s<ValueType>::value, ValueType>::type
+get_rand_value(ValueDistribution &&value_dist, Engine &&gen)
+{
+    return value_dist(gen);
+}
+
+/**
+ * Specialization for complex types.
+ *
+ * @copydoc get_rand_value
+ */
+template <typename ValueType, typename ValueDistribution, typename Engine>
+typename std::enable_if<gko::is_complex_s<ValueType>::value, ValueType>::type
+get_rand_value(ValueDistribution &&value_dist, Engine &&gen)
+{
+    return ValueType(value_dist(gen), value_dist(gen));
+}
+
+/**
+ * timing the apply operation A->apply(b, x). It will runs 2 warmup and get
+ * average time among 10 times.
+ *
+ * @return seconds
+ */
+double timing(std::shared_ptr<const gko::Executor> exec,
+              std::shared_ptr<gko::LinOp> A, std::shared_ptr<gko::LinOp> b,
+              std::shared_ptr<gko::LinOp> x)
```

Comment:
How about using plain pointers here? Then you could get rid of all the `share`s below.

Reply:
alternately, it will use lend when calling the function, right?